packages/components/src/Navbar/Component.purs

```diff
@@ -56,48 +55,47 @@ fromJSProps jsProps =
         "JUNIOR" -> JUNIOR
         _        -> KSF
 
-type State =
-  { collapsedNavVisibility :: Visibility  }
-
-initialState :: State
-initialState = { collapsedNavVisibility: Hidden }
-
-jsComponent :: React.ReactComponent JSProps
-jsComponent = React.toReactComponent fromJSProps component { render }
-
-component :: React.Component Props
-component = React.createComponent "Navbar"
-
-navbar :: Props -> JSX
-navbar = make component
-  { initialState, render }
-
-render :: Self -> JSX
-render self =
+component :: Component Props
+component = React.component "Navbar" $ \ props -> React.do
+  collapsedNavVisibility /\ modifyCollapsedNavVisibility <- useState Hidden
+  let collapsedNavJSX = collapsedNav props collapsedNavVisibility modifyCollapsedNavVisibility
+  pure $ render props collapsedNavJSX modifyCollapsedNavVisibility
```

Comment:
I think this would be easier to read if you'd make these definitions at this location:
```
let logout = do
      props.logout
      modifyCollapsedNavVisibility $ const Hidden
    toggle = modifyCollapsedNavVisibility negateVisiblity
```

And then pass these to the render function instead of `modifyCollapsedNavVisibility`.

It might be a bit too subtle to edit the props you pass to render to contain this new `logout`.

Reply:
I did what you suggested about replacing `modifyCollapsedVisibility`. We could refactor this module more later.